src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -173,7 +173,14 @@ protected override void Dispose(bool disposing)
                         var attr = File.GetAttributes(copiedFile);
                         if (!attr.HasFlag(FileAttributes.Directory))
                         {
-                            File.SetLastWriteTimeUtc(copiedFile, entry.LastWriteTime.UtcDateTime);
+                            try
+                            {
+                                File.SetLastWriteTimeUtc(copiedFile, entry.LastWriteTime.UtcDateTime);
+                            }
+                            catch (ArgumentOutOfRangeException)
```

Comment:
We shouldn't that's a mono bug, that will eventually get fixed. And we shouldn't check on Windows.

We should open the mono bug if it is not there already